lib/ddtrace/contrib/sinatra/patcher.rb

```diff
@@ -18,12 +41,23 @@ def target_version
         def patch
           require 'ddtrace/contrib/sinatra/tracer'
           register_tracer
+
+          patch_default_middlewares
+          setup_tracer
         end
 
         def register_tracer
           ::Sinatra.send(:register, Datadog::Contrib::Sinatra::Tracer)
           ::Sinatra::Base.prepend(Sinatra::Tracer::Base)
         end
+
+        def setup_tracer
+          ::Sinatra::Base.singleton_class.prepend(TracerSetupPatch)
+        end
+
+        def patch_default_middlewares
+          ::Sinatra::Base.singleton_class.prepend(DefaultMiddlewarePatch)
+        end
```

Comment:
Any reason for not implementing the new features from `TracerSetupPatch` and `DefaultMiddlewarePatch` directly in the already existing `Sinatra::Tracer::Base`?

Reply:
I wanted to semantically isolate the different patches instead of making `Sinatra::Tracer::Base` a big monolith.